app/api/services/teams.py

```diff
@@ -0,0 +1,210 @@
+                  .one_or_none())
+
+        return team._asdict() if team else None
+
+    def get_user_teams(self, user_id):
+        result = (
+            db
+            .session
+            .query(
+                Team.id,
+                Team.name,
+                TeamMember.is_team_lead,
+                func.array_agg(cast(TeamMemberPermission.permission, TEXT)).label('permissions')
+            )
+            .join(TeamMember)
+            .join(TeamMemberPermission, isouter=True)
+            .filter(TeamMember.user_id == user_id)
+            .filter(Team.status == 'completed')
+            .group_by(Team.id, Team.name, TeamMember.is_team_lead)
+            .all()
+        )
+        return [r._asdict() for r in result]
+
+    def get_teams_overview(self, user_id, email_domain):
+        teams_led_by_user = (
+            db
+            .session
+            .query(TeamMember.team_id)
+            .join(Team)
+            .filter(
+                Team.status == 'completed',
+                TeamMember.user_id == user_id,
+                TeamMember.is_team_lead.is_(True)
+            )
+            .subquery('teams_led_by_user')
+        )
+
+        team_leads = (
+            db
+            .session
+            .query(
+                TeamMember.team_id,
+                func.array_agg(aggregate_order_by(User.name, User.name)).label('leads')
+            )
+            .join(User)
+            .filter(
+                TeamMember.is_team_lead.is_(True),
+                User.email_address.like('%@{}'.format(email_domain)),
+                User.role == 'buyer'
+            )
+            .group_by(TeamMember.team_id)
+            .subquery('team_leads')
+        )
+
+        team_members = (
+            db
+            .session
+            .query(
+                TeamMember.team_id,
+                func.array_agg(aggregate_order_by(User.name, User.name)).label('members')
+            )
+            .join(User)
+            .filter(
+                TeamMember.is_team_lead.is_(False),
+                User.email_address.like('%@{}'.format(email_domain)),
+                User.role == 'buyer'
+            )
+            .group_by(TeamMember.team_id)
+            .subquery('team_members')
+        )
+
+        result = (
+            db
+            .session
+            .query(
+                Team.id,
+                Team.name,
+                team_leads.columns.leads,
+                team_members.columns.members,
+                case(
+                    [(teams_led_by_user.columns.team_id.isnot(None), True)],
+                    else_=False
+                ).label('isTeamLead')
+            )
+            .join(team_leads, team_leads.columns.team_id == Team.id, isouter=True)
+            .join(team_members, team_members.columns.team_id == Team.id, isouter=True)
+            .join(teams_led_by_user, teams_led_by_user.columns.team_id == Team.id, isouter=True)
+            .filter(Team.status == 'completed')
```

Comment:
I'm not sure if this is the right spot in code, but as a buyer on a domain other than `digital.gov.au`, I'm able to see all the completed teams created by us in the RC environment when I request `/api/2/teams`, which I imagine shouldn't be the case - i.e. this query is not restricting results to teams of other users with the same domain. 

Maybe we should store domain with the team as a column?

Reply:
fixed with inner join to team leads